src/logic/schedule-exporter/workers-absence-export.logic.ts

```diff
@@ -0,0 +1,199 @@
+    workSheet.getRow(1).font = { bold: true };
+  }
+
+  private isNonHolidayWeekDay(verboseDate: VerboseDate): boolean {
+    return !(
+      verboseDate.isPublicHoliday ||
+      verboseDate.dayOfWeek === WeekDay.SA ||
+      verboseDate.dayOfWeek === WeekDay.SU
+    );
+  }
+
+  private pushWorkersRow(
+    name: string,
+    workers: (string | number)[][],
+    shift: ShiftCode,
+    from: number,
+    to: number,
+    month: number,
+    daysNo: number,
+    workerContractType: ContractType,
+    verboseDates: VerboseDate[]
+  ): void {
+    workers.push([
+      name,
+      SHIFTS[shift].name,
+      `${from} ${TranslationHelper.polishMonthsGenetivus[month]}`,
+      `${to} ${TranslationHelper.polishMonthsGenetivus[month]}`,
+      daysNo,
+      WorkerHourInfo.calculateFreeHoursForContractType(
+        workerContractType,
+        this.scheduleModel.shifts[name] as ShiftCode[],
+        this.revision.shifts[name],
+        verboseDates as DateInformationForWorkInfoCalculation[],
+        this.scheduleModel.shift_types
+      ),
+    ]);
+  }
+
+  private setDaysNoAndIndex(
+    index: number,
+    workerShifts: ShiftCode[],
+    verboseDates: VerboseDate[],
+    daysNo: number
+  ): [number, number] {
+    while (index + 1 < workerShifts.length && workerShifts[index + 1] === workerShifts[index]) {
+      workerShifts[index] = ShiftCode.W;
+      index++;
+      if (this.isNonHolidayWeekDay(verboseDates[index])) daysNo++;
+    }
+    return [daysNo, index];
+  }
+
+  private createAbsenceInfoSection(scheduleModel: ScheduleDataModel): AbsenceInfo {
+    const cellsToMerge: number[][] = [];
+    const names = Object.keys(scheduleModel.employee_info?.type);
+    const month = scheduleModel.schedule_info.month_number;
+    const year = scheduleModel.schedule_info.year;
+    const dates = scheduleModel.month_info.dates;
+    const workers: (string | number)[][] = [];
+
+    let employeeRowIndex = 2;
+
+    workers.push(ABSENCE_HEADERS);
+    workers.push(EMPTY_ROW);
+    names.forEach((name) => {
+      const workerShifts = scheduleModel.shifts[name] as ShiftCode[];
+      const workerContractType = scheduleModel.employee_info.contractType[name];
+      const { verboseDates } = new MonthInfoLogic(month, year, dates);
+
+      let moreThanOneRow = false;
+      workerShifts.forEach((shift, index) => {
+        let daysNo = 0;
+        if (this.isNonHolidayWeekDay(verboseDates[index])) daysNo++;
+        if (shift !== ShiftCode.W && shift !== ShiftCode.NZ && !SHIFTS[shift].isWorkingShift) {
+          employeeRowIndex++;
+          const from = scheduleModel.month_info.dates[index];
+          const [totalDaysNo, endIndex] = this.setDaysNoAndIndex(
+            index,
+            workerShifts,
+            verboseDates,
+            daysNo
+          );
+          const to = scheduleModel.month_info.dates[endIndex];
+
+          workerShifts[endIndex] = ShiftCode.W;
+          if (!moreThanOneRow) {
+            this.pushWorkersRow(
```

Comment:
Zawsze wywołujemy tę funkcję tylko `name` jest inny. Zamieniłbym to na ustawianie name'a w if/else i wywołanie funkcji na końcu, będzie czytelniej :D